### ISSUE TITLE
[Suggestion] Bring back the Essentia Distillery (TC4 Advanced Alchemical Furnace)

### ISSUE THREAD

**Issue by Mestrophis** [TheCodex6824/ThaumicAugmentation#247]
I'm sur this is something you've had suggested to you multiple times, but I didn't see anyone mention it here. Basically, I feel like it'd be great to bring back TC4's advanced alchemical furnace in a TA-flavored method. 

The main issue with TC6's essentia distillation mechanics is that, even with a perfectly optimal setup, a portion of essentia is lost as flux. Since the best way to clean up flux is via piping essentia into a flux condenser, this causes players to lose even more essentia than their smeltery would advertise. And, more importantly, it's slow. Essentia takes forever to boil and distill even with lategame setups. Not to mention that automating an alchemical furnace is a pain, since the input side for fuel is at the bottom, something you can't do with hoppers, though you probably can with golems. In short, the current alchemical furnaces are obnoxious to use.

Enter the Essentia Distillery. In Thaumcraft 4, the Advanced Alchemical Furnace was a late-game essentia smelting multiblock which was highly efficient and had a unique method of powering it: running on ignis, aqua, and perditio centivis energy to break down essentia at high speed. Seeing that Thaumic Augmentation has ported over several classic features from previous versions of Thaumcraft, it'd make sense to have the AAC return as an end-game essentia distillation option.

The main draw for me is that the Essentia Distillery could be a perfect practical use of stored impetus. Right now, its only really useful for combat (via the impetus cannon and the void shield spell), a novelty feature (the arcane terraformer), and a mobility tool (impetus thruster for elytra harness). Using impetus to power the distillation process of an advanced alchemical furnace would be a highly-applicable and perfect way to consume stored impetus, using its chaotic essence to violently but efficiently break down objects into their aspects.

Though one thing I am interested in is the maintenance cost. In TC4, the AAC required a constant input of Ignis CV just to keep warmed up, as it wouldn't function otherwise. This usually wasn't a problem as most end-game TC bases have a powerful energized node providing all aspects everywhere. Would this possibly translate into TA as a gradual aura drain? This would be interesting as it'd basically be an excuse to stock up on Vis Batteries and Vis Regenerators. Alternately, redstone could be used to activate/deactivate the Essentia Distillery. And of course the question of if the construct acts as an impetus node (and can be directly connected to) or if it draws raw impetus from a nearby provider (like a flux rift or impetus diffuser).

These are all weird questions and this rambled on a long time, but this is something I'm very excited about and I would like to hear your thoughts on it.

**Comment by Mestrophis** [comment]
First thing: will the machine have 100% essentia efficiency? I think it should, because the "flux tax" dynamic of current essentia smelteries is pretty pointless and only serves to indirectly drain essentia from your reserves.

**Comment by Vapaman** [comment]
I don't think that the "flux tax" is primarly there to drain the essentia, it is just too low to make too much of an impact, unless you are using the basic smeltery. Instead I think it is a sort of a punishing mechanic, where it functions as a motivation to improve the smelting system and plan the uses better.

I think that the main motivation for the Advanced Alchemical Furnace in TC4 was that there were no way of processing items with large amounts of aspects (kind of a niche use), and the slow processing speed of the normal alchemical furnace. I think that if added to Augmentation, the main draw would be 100% essentia distilling efficiency.

Now what would be the consequences of this? At least to me it would make Impetus system a lot more automatable, with no need for the all the time clogging condensers when smelting nether wart into vitium and feeding the vitium into a rift. The system is not be entirely flux free, because when dealing with the extra essentia many resort to void jars, which cause a tiny bit of flux but it would be a lot easier to manage, because you can transport the essentia to be voided in the chunk you want and use a part of it for cleaning the flux.

If Augmentation wants to add more uses for impetus, the system should be entirely reliable when automated properly. A 100% flux efficient vitium essentia output would fix my current issues with it, and it would circumvent the need to use of the condencers without making them enterily useless.

**Comment by Mestrophis** [comment]
100% Efficiency would be amazing. I also think it would be cool that the furnace's output wouldn't be as limited by how quickly essentia can be distilled into alembics and instead can be directly output through pipes. I wonder if it'd be possible to make it output super fast though? I know Flux Inducers directly connected to warded jars can drain vitium at a frightening rate, so would it be possible to do it for this too? Or, if not, perhaps just having multiple outputs would suffice.

**Comment by SillyFrilly** [comment]
Honestly, I'm on board with the idea of making the Advanced Alchemical Furnace impetus- based; the cost of 100% efficiency is the amount of Vitium essentia you need to (indirectly) create the impetus needed to run your systems, right?

You mentioned the possibility of draining the aura while it was active, but isn't there also the possibility of it just having the same sort of drawback that the original machine had in Thaumcraft 4 did (_in that it'd need a constant trickle of Impetus even when not processing anything, or you'd have to wait until it's fully charged to start it up again_)? Maybe it can even distill and output essentia faster the more Impetus you provide it- making it relatively slow _at first_, but becoming a lot more viable if you have an optimized Impetus setup that can handle the expenses of running it at full steam.

**Comment by Mestrophis** [comment]
I mentioned the distinction of aura to maintain / impetus to run, because the vis needed to run the furnace in TC4 was Ignis just to maintain, and Aqua/Perditio to run it. 
I like the idea of having to control and manage the aura near the smelter to keep it running, something about how an endgame machine requires having mastery of the aura system seems pretty cool. 

Though now I'm imagining add-on blocks, like what the normal essentia smelters have. Though with different functions compared to the ones for normal essentia smelters (as aux pumps wouldn't really help with distillation, and aux vents are useless when no essentia is wasted as flux). Dedicated add-ons to make the smelter output faster, for example.

**Comment by SillyFrilly** [comment]
Maybe there could be add-ons that could either lessen its impact on the aura, _or_ reduce its impetus cost, at the price of making the other more expensive? Or maybe an upgrade that allows it to take in specific kinds of essentia?


...Then again, perhaps it'd be better to just give the Auxillary pumps and vents new jobs to do with the new Advanced Alchemical Furnace; the bellows can help speed it up, as it does with the normal smelteries, increasing both the output speed and running costs for the furnace. Maybe the pumps actually take in essentia now, like Auram, which alleviates the aura cost while it chews through Auram essentia (If there is no more Auram, the Advanced Alchemical Furnace [_shouldn't it be AAF?_] acts as though the upgrade wasn't there), and the vents could be some kind of cooling system that allows the machine to run optimally at higher temperatures-- letting it put in more work with the same amount of impetus.

**Comment by Mestrophis** [comment]
I like these

**Comment by Vapaman** [comment]
I really hope that a impetus using Advanced Essentia Smeltery is going to be added in the future. I think it would open up really interesting automation potential and would increase the complexity of this mod and would make impetus production a lot more interesting. Here is a concept that I've had in my mind for a while.
![kuva](https://user-images.githubusercontent.com/77460466/171958824-6abacf54-ae26-4197-a8c3-c534195b85a6.png)
The Advanced Smeltery would pair up with the Essentia Resevoir really nicely as the Smeltery produces a lot of essentia, the player might need a lot more storage space.
